activemodel/lib/active_model/type/helpers/time_value.rb

```diff
@@ -38,6 +42,11 @@ def user_input_in_time_zone(value)
 
         private
 
+          def value_is_rounding_candidate?(value)
+            value.respond_to?(:usec) &&
+              (value.usec.positive? || value.respond_to?(:nsec) && (value.nsec % 1000).positive?)
```

Comment:
Why just not use `value.nsec.positive?`?

Reply:
I suppose I could... For some reason I was thinking, since `nsec` contains the seconds/milliseconds total as well, not just the nanosecond remainder, that I should check that. But, that's an unnecessary calculation in the check, let me revise.